invenio_rdm_records/resources/resources.py

```diff
@@ -295,10 +295,24 @@ def p(route):
         url_rules = [
             route("GET", p(routes["list"]), self.search),
             route("DELETE", p(routes["list"]), self.delete),
+            route("GET", p(routes["count"]), self.count),
         ]
 
         return url_rules
 
+    @request_search_args
+    @request_view_args
+    @response_handler(many=True)
+    def count(self):
```

Comment:
Tests are missing, we should add tests for the resource function and the service function.